assignments/array-methods.js

```diff
@@ -56,28 +56,63 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(runner){
+    fullName.push(`${runner.first_name}, ${runner.last_name}`);//IF THIS IS CORRECT I CANT BELIEVE I FIGURED IT OUT ON THE FIRST TRY!!!!!
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+const makeThemUpper = runners.map(function(runner){
+    allCaps.push(runner.first_name.toUpperCase())
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+const shirtSizeFilter = runners.filter(function(runner){
+    if(runner.shirt_size === "L"){
+        largeShirts.push(runner)
+    }
+})
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+const totalDon = runners.reduce((total, don) => {
+    return total += don.donation;
+}, 0);
+
+ticketPriceTotal.push(totalDon)
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  
+//Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+let allLower = [];
+const makeThemLower = runners.map(function(runner){
+    allLower.push(runner.first_name.toLowerCase())
+})
```

Comment:
If you'd like, you can refactor Problem 1 & 3 to match the patterns I commented about earlier